src/System.Runtime/ref/System.Runtime.cs

```diff
@@ -6139,7 +6139,7 @@ public class TypeDelegator : System.Reflection.TypeInfo
     public delegate bool TypeFilter(Type m, Object filterCriteria);
     public abstract partial class TypeInfo : System.Type, System.Reflection.IReflectableType
     {
-        internal TypeInfo() { }
+        protected TypeInfo() { }
```

Comment:
See https://github.com/dotnet/corefx/issues/14334. We have to either do InternalsVisibleTo or make this public, and only supported on .NET Core. We need to understand and make a call on how we want to support reflection emit across the platforms. Who do we have as the owner of the reflection emit APIs?